pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -27,65 +29,103 @@ import (
 // Node is a set of constraints, compatible pods, and possible instance types that could fulfill these constraints. This
 // will be turned into one or more actual node instances within the cluster after bin packing.
 type Node struct {
-	Constraints         *v1alpha5.Constraints
+	v1alpha5.Constraints
 	InstanceTypeOptions []cloudprovider.InstanceType
 	Pods                []*v1.Pod
-	unschedulable       bool // if true, the node will accept no more pods
+
+	podResources    v1.ResourceList
+	daemonResources v1.ResourceList
 }
 
-func NewNode(constraints *v1alpha5.Constraints, instanceTypeOptions []cloudprovider.InstanceType, pods ...*v1.Pod) *Node {
-	n := &Node{Constraints: constraints.DeepCopy()}
+func NewNode(constraints *v1alpha5.Constraints, daemonResources v1.ResourceList, instanceTypeOptions []cloudprovider.InstanceType) *Node {
+	n := &Node{
+		Constraints:     *constraints.DeepCopy(),
+		daemonResources: daemonResources,
+	}
 
 	for _, it := range instanceTypeOptions {
+		// If a zero-resource pod can't fit, don't consider this instance type.  This occurs if the node overhead +
+		// daemon set requests is already larger than the instance type can support
+		if !n.newPodCanFit(nil, it) {
+			continue
+		}
+
 		// pre-filter our list of all possible instance types by what the provisioner allows
 		if !cloudprovider.Compatible(it, constraints.Requirements) {
 			continue
 		}
 		n.InstanceTypeOptions = append(n.InstanceTypeOptions, it)
 	}
-	// technically we might create some invalid construct here if the pod can't be supported by any instance type
-	// but this is ok as we just won't have any valid instance types, this pod won't be schedulable and nothing else
-	// will be compatible with this node
-	for _, p := range pods {
-		n.Add(p)
-	}
 	return n
 }
 
 func (n Node) Compatible(pod *v1.Pod) error {
```

Comment:
We want compatible to be separate for the next PR so we can implement preferred pod affinity and be a bit smarter about spreading pods around the nodes we already have to create anyway due to hostname topology spreads and pod anti-affinity (e.g. create 3 m5.8xlarge instead of a m5.24xlarge and 2x m5.large).